provider/core/src/resource.rs

```diff
@@ -142,57 +142,46 @@ impl ResourceKey {
         // Regex: [a-zA-Z0-9=_]+(/[a-zA-Z0-9=_]+)*@[0-9]+
         enum State {
             Start,
-            AfterChar,
-            AfterSlash,
-            AfterCharAfterSlash,
-            AfterAt,
-            AfterDigit,
+            Body0,
+            Slash,
+            Body1,
+            AtSign,
+            Body2,
```

Comment:
We can probably argue about names all day. I only made this an enum so the matching is exhaustive. Could even just call them `1,2,3,4,5`